### ISSUE TITLE
Intermittent CORS error due to missing headers

### ISSUE THREAD

**Issue by camilisette** [Azure/azure-functions-python-worker#880]
Experiencing intermittent CORS error. I have an angular app that retrieves data by making multiple calls to different functions in parallel. There are times that the application will throw an error due to CORS. Looking at the response, it looks like the ones that are throwing the error are missing response headers.

I find it strange because all the functions are under a single Function App and there are times when 1 out of 4 requests will throw a CORS error. It's usually resolved by refreshing the page. Then all will load fine. It's intermittent and unpredictable. If I keep refreshing 1 out 5 times, some requests will throw a CORS error.

Also tried allowing all origins (*) through the portal, still facing the same issue. Is this a bug that sometimes response headers are missing?

#### Investigative information

Please provide the following:

- Timestamp: 2021-07-21T20:57:27Z
- Function App version: 3.0.15885.0
- Invocation ID: cf718d6a-f6ae-4ecc-b3bf-173a6e45a6a4
- Region: Southeast Asia

#### Repro steps

Provide the steps required to reproduce the problem:

1. Send a request to different functions in parallel using an Angular App
2. If no error, refresh page a couple times until it throws an error

#### Expected behavior

Provide a description of the expected behavior.

1. Functions should always return response headers

#### Actual behavior

Provide a description of the actual behavior observed.

1. There are times that some functions will not return response headers. When different functions are called in parallel, there are times that some of them will not return response headers. Causing CORS error.

#### Known workarounds

Provide a description of any known workarounds.

1. Refresh page, but this is not a fix.

#### Related information 

Provide any related information 

* Programming language used:
* Web app - Angular with Authentication using MSAL which send Authentication Bearer token on each call to API.
* Functions - Python. Validates the token received from application, if authorized main function logic will be executed.

**Some updates:** Did some more testing today and it seems that it's because we're intermittently getting a 404 on some functions...

**Comment by pragnagopa** [comment]
@Hazhzeng @vrdmr  - Please take a look. From the stack look like issue in the python worker 
```
> Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcException : Result: Failure Exception: UnboundLocalError: local variable 'HEADERS' referenced before assignment Stack:   File '/azure-functions-host/workers/python/3.9/LINUX/X64/azure_functions_worker/dispatcher.py', line 398, in _handle__invocation_request     call_result = await self._loop.run_in_executor(   File '/usr/local/lib/python3.9/concurrent/futures/thread.py', line 52, in run     result = self.fn(*self.args, **self.kwargs)   File '/azure-functions-host/workers/python/3.9/LINUX/X64/azure_functions_worker/dispatcher.py', line 602, in _run_sync_func     return ExtensionManager.get_sync_invocation_wrapper(context,   File '/azure-functions-host/workers/python/3.9/LINUX/X64/azure_functions_worker/extension.py', line 215, in _raw_invocation_wrapper     result = function(**args)   File '/home/site/wwwroot/riders/__init__.py', line 98, in main     return func.HttpResponse(json.dumps(error), status_code=401, headers = HEADERS)
     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Script.Description.WorkerFunctionInvoker.InvokeCore(Object[] parameters,FunctionInvocationContext context) at /src/azure-functions-host/src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs : 93
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Script.Description.FunctionInvokerBase.Invoke(Object[] parameters) at /src/azure-functions-host/src/WebJobs.Script/Description/FunctionInvokerBase.cs : 82
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Script.Description.FunctionGenerator.Coerce[T](Task`1 src) at /src/azure-functions-host/src/WebJobs.Script/Description/FunctionGenerator.cs : 225
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionInvoker`2.InvokeAsync[TReflected,TReturnValue](Object instance,Object[] arguments) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionInvoker.cs : 52
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.InvokeWithTimeoutAsync(IFunctionInvoker invoker,ParameterHelper parameterHelper,CancellationTokenSource timeoutTokenSource,CancellationTokenSource functionCancellationTokenSource,Boolean throwOnTimeout,TimeSpan timerInterval,IFunctionInstance instance) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs : 555
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance,ParameterHelper parameterHelper,ILogger logger,CancellationTokenSource functionCancellationTokenSource) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs : 501
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance,FunctionStartedMessage message,FunctionInstanceLogEntry instanceLogEntry,ParameterHelper parameterHelper,ILogger logger,CancellationToken cancellationToken) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs : 279    End of inner exception
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance,FunctionStartedMessage message,FunctionInstanceLogEntry instanceLogEntry,ParameterHelper parameterHelper,ILogger logger,CancellationToken cancellationToken) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs : 326
    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
    at async Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance functionInstance,CancellationToken cancellationToken) at C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs : 94
```

**Comment by camilisette** [comment]
This error was a coding issue on my end. I was trying to return some headers if 'origin' was one of the request headers. But I put the 'HEADERS' declaration inside an if clause. So when I tried to reach the function from localhost which didn't give 'origin' as a request header, it threw this error.

Nonetheless, issue is still there after I fixed that.

So far the issue hasn't reoccurred after I implemented the workaround of checking availability of all the functions every minute (to keep them running). So it must be related to the the functions go to sleep.
